server/context.go

```diff
@@ -204,6 +211,18 @@ func (ctx *Context) parseGossipBootstrapResolvers() ([]gossip.Resolver, error) {
 	return bootstrapResolvers, nil
 }
 
+// GetHTTPClient returns the context http client, initializing it
+// if needed. It uses the context Certs.
+// TODO(marc): are there odd cases where may need a mutex?
```

Comment:
If this races it could cause problems beyond simply creating two clients (https://software.intel.com/en-us/blogs/2013/01/06/benign-data-races-what-could-possibly-go-wrong). Just use a Mutex (or an atomic unsafe pointer if you must, although I'm kind of wary of our increasing use of that pattern).